src/java/htsjdk/samtools/SAMRecordDuplicateComparator.java

```diff
@@ -200,6 +204,10 @@ private boolean pairedEndAndBothMapped(final SAMRecord record) {
     public int compare(final SAMRecord samRecord1, final SAMRecord samRecord2) {
         int cmp;
 
+        if (null == samRecord1.getHeader() || null == samRecord2.getHeader()) {
+            throw new IllegalArgumentException("Record must have non-null SAMFileHeaders to be comapred");
```

Comment:
Also this check is redundant, since the first thing `compare()` does is to call `fileOrderCompare()`